public/src/modules/sounds.js

```diff
@@ -1,90 +1,89 @@
 "use strict";
 /* global app, define, socket, config */
 
-define('sounds', ['buzz'], function (buzz) {
+define('sounds', function () {
 	var	Sounds = {};
 
-	var loadedSounds = {};
-	var eventSoundMapping;
-	var files;
+	var fileMap;
+	var soundMap;
+	var cache = {};
 
-	socket.on('event:sounds.reloadMapping', function () {
-		Sounds.reloadMapping();
-	});
-
-	Sounds.reloadMapping = function () {
-		socket.emit('modules.sounds.getMapping', function (err, mapping) {
+	Sounds.loadMap = function loadMap(callback) {
+		socket.emit('modules.sounds.getUserSoundMap', function (err, map) {
 			if (err) {
 				return app.alertError(err.message);
 			}
-			eventSoundMapping = mapping;
+			soundMap = map;
+			if (callback) {
+				callback();
+			}
 		});
 	};
 
 	function loadData(callback) {
-		socket.emit('modules.sounds.getData', function (err, data) {
-			if (err) {
-				return app.alertError('[sounds] Could not load sound mapping!');
-			}
-			eventSoundMapping = data.mapping;
-			files = data.files;
-			callback();
-		});
-	}
-
-	function isSoundLoaded(fileName) {
-		return loadedSounds[fileName];
-	}
-
-	function loadFile(fileName, callback) {
-		function createSound() {
-			if (files && files[fileName]) {
-				loadedSounds[fileName] = new buzz.sound(files[fileName]);
+		var outstanding = 2;
+		function after() {
+			outstanding -= 1;
+			if (outstanding === 0 && callback) {
+				callback();
 			}
-			callback();
 		}
-
-		if (isSoundLoaded(fileName)) {
-			return callback();
+		if (fileMap) {
+			outstanding -= 1;
+		} else {
+			$.getJSON(config.relative_path  + '/assets/sounds/fileMap.json', function (map) {
+				fileMap = map;
+				after();
+			});
 		}
 
-		if (!files || !files[fileName]) {
-			return loadData(createSound);
-		}
-		createSound();
+		Sounds.loadMap(after);
 	}
 
-	Sounds.play = function (name) {
-		function play() {
-			Sounds.playFile(eventSoundMapping[name]);
+	Sounds.playSound = function playSound(soundName) {
+		if (!soundMap || !fileMap) {
+			return loadData(after);
 		}
 
-		if (!eventSoundMapping) {
-			return loadData(play);
+		function after() {
+			if (!fileMap[soundName]) {
+				return;
+			}
+			var audio = cache[soundName] = cache[soundName] || new Audio(config.relative_path + '/assets/sounds/' + fileMap[soundName]);
+			audio.pause();
+ 			audio.currentTime = 0;
+			audio.play();
 		}
 
-		play();
+		after();
 	};
 
-	Sounds.playFile = function (fileName) {
-		if (!fileName) {
-			return;
-		}
-
-		function play() {
-			if (loadedSounds[fileName]) {
-				loadedSounds[fileName].play();
-			} else {
-				app.alertError('[sounds] Not found: ' + fileName);
+	Sounds.play = function play(type, id) {
+		function after() {
+			if (!soundMap[type]) {
+				return;
+			}
+			
+			if (id) {
+				if (localStorage.getItem('sounds.handled:' + id)) {
+					return;
+				}
+				localStorage.setItem('sounds.handled:' + id, true);
```

Comment:
This theoretically prevents multiple tabs from playing audio for the same notification / chat message.

I did actually forget something: I meant to clear that field after a timeout. And yes, sessionstorage is probably more fitting here, as well.